packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Next,
+  Middleware,
+  MiddlewareRequest,
+  MiddlewareResponse,
+} from '../types/types'
+
+export default function createConcurrentModificationMiddleware(): Middleware {
+  return (next: Next) => {
+    return async (request: MiddlewareRequest): Promise<MiddlewareResponse> => {
+      const response = await next(request)
+      if (response.statusCode == 409) {
+        /**
+         * extract the currentVersion
+         * from the error body and update
+         * request with the currentVersion
+         */
+        const version = response.error?.body?.errors?.[0]?.currentVersion
+
+        // update the resource version here
+        request.body =
```

Comment:
I would add an if-statement and only retry if the version is set

Pseudo-Code:
```
const version =

if (version) {
   // update request
  return next(request)
}
return response
```